clusterctl/cmd/validate_cluster.go

```diff
@@ -34,9 +48,113 @@ var validateClusterCmd = &cobra.Command{
 }
 
 func init() {
+	validateClusterCmd.Flags().StringVarP(
+		&vco.ClusterName, "cluster-name", "", "",
+		"The name of the cluster to be validated. This can be omitted if there is only one cluster being managed in the given namespace.")
+	validateClusterCmd.Flags().StringVarP(
+		&vco.Namespace, "namespace", "n", v1.NamespaceDefault,
+		"If present, validate the cluster in the specified namespace.")
 	validateCmd.AddCommand(validateClusterCmd)
 }
 
 func RunValidateCluster() error {
-	return errors.NotImplementedError
+	config, err := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
+		clientcmd.NewDefaultClientConfigLoadingRules(),
+		&clientcmd.ConfigOverrides{}).ClientConfig()
+	if err != nil {
+		return fmt.Errorf("Failed to create client for talking to the apiserver: %v", err)
+	}
+
+	client, err := clientset.NewForConfig(config)
+	if err != nil {
+		return fmt.Errorf("Failed to create client for talking to the apiserver: %v", err)
+	}
+
+	cluster, err := getClusterObject(client, vco.ClusterName, vco.Namespace)
+	if err != nil {
```

Comment:
cuddle this and everywhere else

```
if err := foo(); err != nil {

}
```

Reply:
At this place, we need |cluster| outside of the if scope. I have cuddled other places where only err is returned.